### PR TITLE
feat: Adds support for showing navbar in embedded view.

### LINKED ISSUE #23559: Show Navbar in template detailed view when `?embed=true`
We came accross this issue while publishing: https://app.appsmith.com/templates/646624e5b322f947ee5ce916

We noticed that in the template detailed view due to presence of `?embed=true` it was stripping navbar from the preview. And when we somehow forcibly put a non embedded url to the iFrame it had a lot of nonessential options such as:
1. Fork app option
2. User profile option
3. Sign out option

We need a way to have navbar after stripping above features, with URL like : `?embed=true&navbar=true`

- [x] Make view functional with Navbar
- [x] Add a toggle option in `App settings` to show navbar

### PR DESCRIPTION
**Pull request by rahulbarwal**
## Description

- Adds support for query param navbar. We will show navbar and user settings in the embedded view based on this.
- Made sure that user profile settings are not visiblie in embedded view when navbar=true
- Adds logic to let user decide Template iframe URL. Till now, we forced `embed=true` flag, moving forward we let this on user(devrel's) descretion.
- Adds correct url for embed settings tab.

- Adds relevant tests for both embed settings and appview.

#### PR fixes following issue(s)
Fixes #23559

#### Media

| App normal preview   |     ?embed=true      |  ?embed=true&navbar=true |
|----------|:-------------:|------:|
| Screenshot 2023-05-26 at 3 08 16 PM |  Screenshot 2023-05-26 at 3 08 39 PM | Screenshot 2023-05-26 at 3 08 59 PM |

#### Type of change
- New feature (non-breaking change which adds functionality)

## Testing
- [ ] Cypress
#### Dev activity
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Test-plan-implementation#speedbreaker-features-to-consider-for-every-change) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans/_edit#areas-of-interest)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

### CODE REVIEW

**Review by akash-codemonk — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ViewMode_spec.js [hunk 1]
```diff
@@ -1,18 +1,53 @@
...
   it("1. on click of apps on header, it should take to application home page", function () {
-    cy.PublishtheApp();
+    cy.get(appNavigationLocators.userProfileDropdownButton).should("exist");
     cy.get(
       `${appNavigationLocators.header} ${appNavigationLocators.backToAppsButton}`,
     ).click();
     cy.url().should("eq", BASE_URL + "applications");
   });
+
+  it("2. In the published app with embed=true, there should be no header", function () {
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ViewMode_spec.js:34]: When embed=true & no navbar defined, its default value is false?

Comment (rahulbarwal) [reply app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ViewMode_spec.js:34]: @Aishwarya-U-R 
Yes, well not exactly `false`, but `undefined` which is treated as `false`.

File: app/client/cypress/support/Pages/InviteModal.ts [hunk 2]
```diff
@@ -79,12 +79,7 @@ export class InviteModal {
     this.SelectEmbedTab();
     this.embedSettings.ToggleShowNavigationBar(toShowNavBar);
     cy.get(this.locators._previewEmbed).invoke("removeAttr", "target").click();
-    if (toShowNavBar === "true") {
-      this.agHelper.AssertElementExist(this.commonLocators._backToEditor);
-      this.deployPage.NavigateBacktoEditor();
-    } else {
-      this.agHelper.AssertElementAbsence(this.commonLocators._backToEditor);
-      cy.go("back");
-    }
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/InviteModal.ts:88]: Why is this removed?

File: app/client/cypress/support/Pages/AppSettings/EmbedSettings.ts [hunk 3]
```diff
@@ -1,4 +1,6 @@
 import { ObjectsRegistry } from "../../Objects/Registry";
+import HomePage from "../../../locators/HomePage";
```
Comment (Aishwarya-U-R) [inline app/client/cypress/support/Pages/AppSettings/EmbedSettings.ts:2]: Remove unused imports.

Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AppSettings/EmbedSettings.ts:2]: Even if HomePage is needed, it should be used from ObjectsRegistry.

Comment (rahulbarwal) [reply app/client/cypress/support/Pages/AppSettings/EmbedSettings.ts:2]: Done.

File: app/client/cypress/support/Pages/InviteModal.ts (hunk 2 above)
Comment (rahulbarwal) [reply app/client/cypress/support/Pages/InviteModal.ts:88]: Because the behavior has changed. 
There are 3 "views" now:
1. Deployed view
2. Deployed view with embed=true
3. Deployed view with embed=true&navbar=true

We will get `NavigateBackToEditor` only in `1`. When embed=true the back to editor won't be visible.

**Review by Aishwarya-U-R — COMMENTED** [review]
Cypress looks good.

**Review by dhruvikn — DISMISSED** [review]
The PR looks good to me. 
Just generating a DP to try this out once.

File: app/client/cypress/support/Pages/AppSettings/EmbedSettings.ts (hunk 3 above)
Comment (Aishwarya-U-R) [reply app/client/cypress/support/Pages/AppSettings/EmbedSettings.ts]: @dhruvikn - It was done. Do you still see unused imports?

File: app/client/src/pages/AppViewer/Navigation/Sidebar.tsx [hunk 4]
```diff
@@ -210,40 +211,42 @@ export function Sidebar(props: SidebarProps) {
...
+            <StyledCtaContainer>
+              <ShareButton
+                currentApplicationDetails={currentApplicationDetails}
+                currentWorkspaceId={currentWorkspaceId}
+                insideSidebar
+                isMinimal={isMinimal}
+              />
+
+              <PrimaryCTA
+                className="t--back-to-editor"
```
Comment (tanvibhakta) [inline app/client/src/pages/AppViewer/Navigation/Sidebar.tsx:226]: can we use data-testid?

**Review by tanvibhakta — DISMISSED** [review]

**Comment by dhruvikn** [comment]
Hey @rahulbarwal, when can we merge this? I am working on [an issue](https://github.com/appsmithorg/appsmith/issues/23190) that will be affected by this PR.

**Comment by rahulbarwal** [comment]
@dhruvikn I am waiting for QA to add test plan approved label. Otherwise it is ready.
CC: @shwetha-ramesh

**Comment by dhruvikn** [comment]
Please let me know once it's merged. Thanks! @rahulbarwal

**Comment by shwetha-ramesh** [comment]
@rahulbarwal. Qa done

File: app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ViewMode_spec.js [hunk 5]
```diff
@@ -1,18 +1,53 @@
...
+});
+
+Cypress.Commands.add("getSharedUrl", () => {
+  return Cypress.sharedStore.url;
+});
+
 describe("Preview mode functionality", function () {
   before(() => {
     cy.addDsl(dsl);
+    cy.PublishtheApp();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/OtherUIFeatures/ViewMode_spec.js:18]: Please use TS helpers here instead.

**Review by akash-codemonk — APPROVED** [review]

File: app/client/cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts [hunk 6]
```diff
@@ -78,4 +78,19 @@ describe("In-app embed settings", () => {
     ValidateSyncWithInviteModal("true");
     ValidateSyncWithInviteModal("false");
   });
+
+  it("5. Changing the show navigation bar setting in the App settings pane should update the embed URL", () => {
+    cy.reload();
```
Comment (Aishwarya-U-R) [inline app/client/cypress/e2e/Regression/ClientSide/SettingsPane/EmbedSettings_spec.ts:83]: Please use TS helpers here instead.

**Review by Aishwarya-U-R — APPROVED** [review]
Approving now. Review comments to be corrected in next PR.
